language_service/src/hover.rs

```diff
@@ -158,12 +188,24 @@ impl<'a> Handler<'a> for HoverGenerator<'a> {
         &mut self,
         path: &'a ast::Path,
         _: &'_ hir::ItemId,
-        _: &'a hir::Package,
+        item: &'a hir::Item,
+        package: &'a hir::Package,
         _: &'a hir::Ident,
         udt: &'a hir::ty::Udt,
     ) {
-        let contents = markdown_fenced_block(self.display.hir_udt(udt));
+        let ns = item
+            .parent
+            .and_then(|parent_id| package.items.get(parent_id))
+            .map_or_else(
+                || Rc::from(""),
+                |parent| match &parent.kind {
+                    hir::ItemKind::Namespace(namespace, _) => namespace.name(),
+                    _ => Rc::from(""),
+                },
+            );
```

Comment:
Can we refactor this into `get_namespace_name` or something , and make the one in `at_callable_ref` call that too?